jbmc/unit/java_bytecode/java_bytecode_parse_generics/parse_bounded_generic_inner_classes.cpp

```diff
@@ -68,6 +68,28 @@ SCENARIO(
             to_struct_type(class_symbol.type), "elem");
         require_type::require_java_generic_parameter(
           elem.type(), boundedinner_name + "::NUM");
+
+        std::string method_name =
+          boundedinner_name + ".f:(Ljava/lang/Number;)V";
+        REQUIRE(new_symbol_table.has_symbol(method_name));
+        THEN("The method parameter type should respect its bound")
+        {
+          const symbolt &method_symbol =
+            new_symbol_table.lookup_ref(method_name);
+          const code_typet &method_type =
+            require_type::require_code(method_symbol.type);
+          const code_typet::parametert &param =
+            require_type::require_parameter(method_type, "x");
+          const typet &param_type = param.type();
+          REQUIRE(param_type.id() == ID_pointer);
+          REQUIRE(param_type.subtype().id() == ID_symbol);
+#if 0
+          // Not currently working as the bound on type variable is ignored
+          REQUIRE(
+            id2string(to_symbol_type(param_type.subtype()).get_identifier()) ==
+            "java::java.lang.Number");
```

Comment:
Shouldn't this work? By using descriptor rather than signature, the function is non-generic with argument `x` of type `Number` (the `method_name` above suggests so too). Before your changes, `x` was a generic parameter `T` with bound `Object`. In the ideal (future) case, when we do parse bounds, `x` will be generic parameter `T` with bound `Number`. Do I understand this correctly?

This applies to the other tests too.

Reply:
In this case it doesn't work because the generic type and its bound are defined on the class , not on the method, so the changes I've made don't affect it, and it uses the method signature rather than the method descriptor. Apart from that your understanding is correct.